src/components/Zoombar.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+    Button,
+    SliderTrack,
+    Slider,
+    SliderMark,
+    SliderFilledTrack,
+    SliderThumb,
+    VStack,
+} from "@chakra-ui/react"
+import { useState } from "react"
+import { GrAddCircle, GrSubtractCircle } from "react-icons/gr"
+import "./Zoombar.css"
+
+type ZoomBarProps = {
+    cellDimension: number
+    setCellDimension: (num: number) => void
+    cellFontSize: number
+    setCellFontSize: (num: number) => void
+}
+
+const ZoomBar = ({
+    cellDimension,
+    setCellDimension,
+    cellFontSize,
+    setCellFontSize,
+}: ZoomBarProps) => {
+    const [sliderMax, sliderStep] = [200, 10]
+    const [sliderValue, setSliderValue] = useState(150) // 150 corresponds to 100% zoom.
+    const [sliderMoving, setSliderMoving] = useState(false)
+
+    const zoomChange = (direction: number) => {
+        if (direction > 0) {
+            if (sliderValue !== sliderMax) {
+                setSliderValue(sliderValue + 10)
+                setCellDimension(cellDimension + 10)
+                setCellFontSize(cellFontSize + 1.5)
+            }
+        } else if (direction < 0) {
+            if (sliderValue !== 0) {
+                setSliderValue(sliderValue - 10)
+                setCellDimension(cellDimension - 10)
+                setCellFontSize(cellFontSize - 1.5)
+            }
+        }
+        setSliderMoving(true)
+        setTimeout(() => setSliderMoving(false), 1000)
+    }
+    return (
+        <VStack p="1" gap="2">
+            <Button onClick={() => zoomChange(1)}>
+                <GrAddCircle size="25px" />
+            </Button>
+            <Button onClick={() => zoomChange(-1)}>
+                <GrSubtractCircle size="25px" />
+            </Button>
+            <Slider
+                h="250px"
+                w="34px"
+                mt="2"
+                aria-label="slider-ex-1"
+                orientation="vertical"
+                value={sliderValue}
+                min={0}
+                max={sliderMax}
+                step={sliderStep}
+                onChange={(val) => {
+                    setSliderValue(val)
+                    setSliderMoving(true)
+                }}
+                onChangeEnd={(val) => {
+                    setCellDimension(val)
+                    setCellFontSize(val / 7.5)
```

Comment:
General comment, no change requested. Is there a particular reason why we have separate state for `fontSize` and `cellDimension`? you could have only `cellDimension` and replace and define

```ts
const cellFontSize = () => cellDimension/7.5
```

Reply:
Nice, I think I will clean that up. It was a happy accident that the font size was a nice ratio of the cell dimension, lol

Reply:
That actually improved performance a little bit maybe.